tests/numpy/test_misc.py

```diff
@@ -129,6 +129,48 @@ def test_all_alltrue_any(nps_app_inst):
         assert nps.any(nps_array).dtype is np.bool
 
 
+def test_array_eq(nps_app_inst):
+    import nums.numpy as nps
+    assert nps_app_inst is not None
+
+    int_array_1 = np.array([[1, 2, 3], [4, 5, 6]])
+    int_array_2 = np.array([[3, 9, 1], [8, 4, 2]])
+    bool_array_1 = np.array([[True, False, True], [True, False, True]])
+    bool_array_2 = np.array([[False, False, True], [False, False, True]])
+    float_array_1 = np.array([[1e10, 1e-8, 1e-8], [1e10, 1e-8, 1e-8]])
+    float_array_2 = np.array([[1.00001e10, 1e-9, 1e-9], [1.00001e10, 1e-9, 1e-9]])
+
+    checks = [(int_array_1, int_array_2),
+              (bool_array_1, bool_array_2),
+              (float_array_1, float_array_2)]
+
+    for check in checks:
+        nps_array_1 = nps.array(check[0]).reshape(block_shape=(2, 2))
+        nps_array_2 = nps.array(check[1]).reshape(block_shape=(2, 2))
+        assert nps.array_equal(nps_array_1, nps_array_1).get() == np.array_equal(check[0], check[0])
+        assert nps.array_equal(nps_array_1, nps_array_2).get() == np.array_equal(check[0], check[1])
+        assert nps.array_equiv(nps_array_1, nps_array_1).get() == np.array_equiv(check[0], check[0])
+        assert nps.array_equiv(nps_array_1, nps_array_2).get() == np.array_equiv(check[0], check[1])
+        assert nps.allclose(nps_array_1, nps_array_1).get() == np.allclose(check[0], check[0])
+        assert nps.allclose(nps_array_1, nps_array_2).get() == np.allclose(check[0], check[1])
+
+        assert nps.array_equal(nps_array_1, nps_array_2).dtype is np.bool
+        assert nps.array_equiv(nps_array_1, nps_array_2).dtype is np.bool
+        assert nps.allclose(nps_array_1, nps_array_2).dtype is np.bool
+
+    # False interaction test
+    false_array_1 = np.array([[False, False], [0, 0], [0.0, 0.0]])
```

Comment:
When this is declared, I believe the resulting array will all be of a single type (e.g. float64). Can you separate the different types into different arrays? bool in one, int in another, etc?